bundle/jsky.app.ot/src/main/java/jsky/app/ot/tpe/TpeEditorTools.java

```diff
@@ -123,12 +111,10 @@ public void actionPerformed(ActionEvent e) {
         });
     }
 
-    private void modeChange(TpeMode mode, Option<Object> arg) {
-        for (TpeImageFeature feat : _tpe.getFeatures()) {
-            if (feat instanceof TpeModeSensitive) {
-                ((TpeModeSensitive) feat).handleModeChange(mode, arg);
-            }
-        }
+    private void modeChange(final TpeMode mode, final Option<Object> arg) {
+        _tpe.getFeatures().stream()
```

Comment:
There are some pretty cool features in Java 8. You can turn almost any collection into a stream, and then if you wish, automatically have elements of that stream further processed in parallel.

If you're interested, a lot of the really nice features of Java 8 are available in a small and easily skimmable book called "Java 8 for the Really Impatient."

(I can send you a PDF if interested.)